### PR TITLE
More efficient ADAPChromatogramBuilder

### PR DESCRIPTION
**Pull request by robinschmid**
I tried to optimize a bit the ADAPChromatogramBuilder and reduce the created objects. 

- optimized the "list all data points of all scans" part
- removed unused arrays
- maybe optimized the add zeros method (at least it feels optimized)
- clean up methods

Seems to work fine for my LC-MS test data.

### CODE REVIEW

**Comment by SteffenHeu** [comment]
Looks good.

What do you think about removing the area calc from ADAPChromatogram and using
`FeatureDataUtils.recalculateIonSeriesDependingTypes(ModularFeature feature)` 
everywhere to make it consistent across all modules?

**Comment by robinschmid** [comment]
Is it already redundant?

**Comment by SteffenHeu** [comment]
Don't think so, but we could put it in the FeatureConvertor.

File: src/main/java/io/github/mzmine/util/FeatureConvertors.java [hunk 1]
```diff
@@ -131,20 +128,12 @@ static public ModularFeature ADAPChromatogramToModularFeature(
...
-    Range<Double> mzRange = Range.closed(chromatogram.getRawDataPointsMZRange().lowerEndpoint(),
-        chromatogram.getRawDataPointsMZRange().upperEndpoint());
-    Range<Float> intensityRange =
-        Range.closed(chromatogram.getRawDataPointsIntensityRange().lowerEndpoint().floatValue(),
-            chromatogram.getRawDataPointsIntensityRange().upperEndpoint().floatValue());
-    modularFeature.set(MZRangeType.class, mzRange);
-    modularFeature.set(RTRangeType.class, rtRange);
-    modularFeature.set(IntensityRangeType.class, intensityRange);
+    // recalculate data dependet types
+    FeatureDataUtils.recalculateIonSeriesDependingTypes(modularFeature, CenterMeasure.AUTO);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/util/FeatureConvertors.java:132]: Auto is not implemented in the math utils

```
 public static double calcCenter(CenterMeasure measure, double[] values, double[] weights,
      Weighting weightTransform, Double noiseLevel, Double maxWeight) {
    switch (measure) {
      case AVG:
        return calcWeightedAvg(values, weights, weightTransform, noiseLevel, maxWeight);
      case MEDIAN:
        return calcMedian(values);
      default:
        return Double.NaN;
    }
  }
```

Comment (robinschmid) [reply src/main/java/io/github/mzmine/util/FeatureConvertors.java:132]: oh

Comment (robinschmid) [reply src/main/java/io/github/mzmine/util/FeatureConvertors.java]: used the default then

**Pull request merged**